README.md

```diff
@@ -86,20 +89,48 @@ function onChange({ rgb, hex }) {
 ###### *Use the `onClick` prop which gets passed down.*
 *Example:*
 ```
-const Button = ({ onClick }) => <button className="btn" onClick={onClick}>My custom button</button>
+const Button = ({ onClick }) => 
+    <button className="btn" onClick={onClick} >My custom button</button>
 <!-------->
 <EyeDropper customComponent={Button} />
 ```
 
-# <a name="api-passThrough"></a>
-#### **passThrough**
+# <a name="api-customProps"></a>
+#### **customProps**
+###### Expects `object`
+###### Requires customComponent to be set
+###### *User can pass in their own data to the customComponent, the data can then be retrieved along with the color values in the onChange handler.*
+*Example:*
+```
+const onChange = ({ rgb, hex, customProps }) => {
+    const { data1, data2, } = customProps
+}
+<!-------->
+<Eyedropper customComponent={Button} customProps={{data1, data2, data3}} onChange={onChange}/>
+```
+
+# <a name="api-colorsPassThrough"></a>
+#### **colorsPassThrough**
 ###### Expects `string`
-###### *If you use your own button component, you can choose to pass down the resulting colors as a prop named by whatever you pass in here.*
+###### *Provides access to the picked color value object { rgb, hex } for the eyedropper component.*
+###### *Name provided here will be the name of the color object*
 *Example:*
 ```
-const Button = ({ onClick, droppedColors }) => <button className="btn" onClick={onClick}>My custom button</button>
+const Button = ({ onClick, pixelColors }) => 
+    <button className="btn" onClick={onClick} style={{backgroundColor: pixelColors.hex}}>My custom button</button>
 <!-------->
-<EyeDropper customComponent={Button} passThrough='droppedColors' />
+<EyeDropper customComponent={Button} colorsPassThrough='pixelColors' />
+```
+
+# <a name="api-disabled"></a>
+#### **disabled**
+###### Expects `boolean`
+###### *Internal property provided by the eyedropper component for passing down to the customComponent. It gives control disabling the button element while color picking is active*
```

Comment:
Hmm... Do you think it's possible to make this more clear?
```suggestion
###### *Allows you to control the `disabled` state of the button.*
```

Reply:
The "disabled" property is declared inside the eyedropper's source, that's why I said it's internal property.

Reply:
Ah! I see.
Then let's make the current implementation of `disabled` button the default, and then allow users to opt-in by passing a boolean they themselves will control. I can take that task upon myself. Let's update this text for my update.

Reply:
The implementation is there already. It is default to true and user can change it by setting in the eyedropper component.

Reply:
Oh! I missunderstood you then. Mb.
I would then argue that my suggested change is more clear in terms of functionality. :smile:

Reply:
Okay I think you can take out my first sentence because users don't really need to know how the internals work.

Reply:
But wait... Let's say the user want to decide for himself completely **when** disabled button should happen. I dont think your implementation allows for that use-case.

Reply:
I think you're right, I only exposed the disabled option in the onClick function. Users should also be able to pass down their own as property in the EyeDropper component.

Well though the users can pass the value into the customProps object which will also work.

Reply:
Yeah! So let's adjust the implementation slightly to support this. :smiley: It's really nice to have.